docs/email/postfix/postfix-spf-dkim-debian-8.md

```diff
@@ -0,0 +1,376 @@
+
+4.  Create the key table `/etc/opendkim/key.table`. It needs to have one line per short domain name in the signing table. Each line will look like:
+
+{: .file-excerpt}
+/etc/opendkim/key.table
+:   ~~~ text
+    example     example.com:YYYYMM:/etc/opendkim/keys/example.private
+    ~~~
+
+Replace `example` with the `example` value you used for the domain in the signing table (make sure to catch the second occurrence at the end where it's followed by `.private`), replace `example.com` with your domain name, and replace the `YYYYMM` with the current 4-digit year and 2-digit month (this is referred to as the selector). The first field connects the signing and key tables. The second field is broken down into 3 sections separated by colons. The first section is the domain name the key is for, the second section is a selector used when looking up key records in DNS, and the third section names the file containing the signing key for the domain.
+
+{: .note}
+>
+> The flow for DKIM lookup starts with the sender's address. The signing table is scanned until an entry whose pattern (first item) matches the address is found, then the second item's value is used to locate the entry in the key table whose key information will be used. For incoming mail the domain and selector are then used to find the public key TXT record in DNS and that public key is used to validate the signature. For outgoing mail the private key is read from the named file and used to generate the signature on the message.
+
+5.  Create the trusted hosts file `/etc/opendkim/trusted.hosts`. It's contents need to be:
+
+{: .file}
+/etc/opendkim/trusted.hosts
+:   ~~~ text
+    127.0.0.1
+    ::1
+    localhost
+    myhostname
+    myhostname.example.com
+    example.com
+    ~~~
+
+When creating the file, change `myhostname` to the name of your server and replace `example.com` with your own domain name. We're identifying the hosts that users will be submitting mail through and that should have their outgoing mail signed, which for basic configurations will just be your own mail server.
+
+6.  Generate keys for each domain. First do a `cd /etc/opendkim/keys`. Then issue the following command:
+
+    opendkim-genkey -b 2048 -r -s YYYYMM
+
+replacing YYYYMM with the current year and month as in the key table. This will give you two files, `YYYYMM.private` containing the key and `YYYYMM.txt` containing the TXT record you'll need to set up DNS. Rename the files so they have names matching the third section of the second field of the key table for the domain:
+
+    mv YYYYMM.private example.private
+    mv YYYYMM.txt example.txt
+
+Repeat the commands in this step for every entry in the key table. The `-b 2048` indicates the number of bits in the RSA key pair used for signing and verification. 1024 bits is the minimum, but with modern hardware 2048 bits is safer and it's possible 4096 bits will be required at some point.
```

Comment:
RFC 6373 section 3.3.3 (http://www.ietf.org/rfc/rfc6376.txt). 2048 bits is the largest key verifiers are required to support, signatures from 4096-bit keys may not fit in a UDP DNS response packet and the verification starts to load up the mail server. The small key size is why key rotation's needed and why they recommend a monthly rotation.